lib/CL/devices/cuda/pocl-cuda.c

```diff
@@ -2389,18 +2389,11 @@ pocl_cuda_submit_node (_cl_command_node *node, cl_command_queue cq, int locked)
         {
           for (unsigned int i = 0; i < cmd->svm_free.num_svm_pointers; i++)
             {
-              void *ptr = cmd->svm_free.svm_pointers[i];
-              POCL_LOCK_OBJ (event->context);
-              pocl_raw_ptr *item = pocl_raw_ptr_set_lookup_with_vm_ptr (
-                event->context->raw_ptrs, ptr);
-              assert (item);
-              pocl_raw_ptr_set_erase (event->context->raw_ptrs, item);
-              POCL_UNLOCK_OBJ (event->context);
-              // Leads to 'undefined symbol: POclReleaseContext'
-              // POname (clReleaseContext) (event->context);
-
-              dev->ops->svm_free (dev, ptr);
-            }
+	      void *ptr = cmd->svm_free.svm_pointers[i];
+	      /* This updates bookkeeping associated with the 'ptr'
+		 done by the PoCL core. */
+	      POname (clSVMFree) (event->context, ptr);
```

Comment:
I have a vague feeling it's not good to call API functions from the driver layer due to some troubles with the ICD loader or such, but cannot remember exactly. Maybe @Kerilk remembers?

Reply:
AFAIK calling via `POname(clFunc)` is equivalent to calling `POclFunc` which always resolves to libpocl symbol, IOW it never goes through the ICD loader (only calling `clFunc` without `PO` would). We're already doing this in a few places in the driver layer (mainly `clRelease*` calls).

Reply:
For reference here is the code of the macros:
https://github.com/pocl/pocl/blob/c72d4f3cd4b8892890f55b2d03a5e64e9e6ead2d/lib/CL/pocl_cl.h#L260-L299